### PR TITLE
parser:  Refine LoadDataStmt

### PR DESCRIPTION
**Pull request by spongedu**
### What problem does this PR solve? 

1.  Add restore part for  #274
2. As is described in [MySQL Manual](https://dev.mysql.com/doc/refman/5.7/en/load-data.html),  `Load DATA` statement support an mix of `ColumnName` and `UserVariable` after the `IGNORE number` clause. 
For example, such sql as:
```sql
LOAD DATA INFILE 'file.txt'
  INTO TABLE t1
  (column1, @dummy, column2, @dummy, column3);
```
is legal in MySQL but not in TiDB for now

### What is changed and how it works?
Refine related part in `parser.y` according to MySQL's implementation. 

Note that this PR just  fix the parser side behavior. More work should be done in `TiDB` to make these behavior right. 

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to be included in the release note

### CODE REVIEW

**Comment by spongedu** [comment]
@kennytm @tiancaiamao PTAL

**Comment by tiancaiamao** [comment]
Please update master @spongedu 
PTAL @jackysp

**Comment by spongedu** [comment]
conflicts resolved. PTAL @tiancaiamao

**Comment by tiancaiamao** [comment]
Conflict again... with your previous PR @spongedu

File: parser_test.go [hunk 1]
```diff
@@ -458,7 +458,10 @@ func (s *testParserSuite) TestDMLStmt(c *C) {
...
-		{"load data local infile '/tmp/t.csv' into table t fields terminated by 'ab' escaped by '' enclosed by 'b' SET b = CAST(CONV(MID(@var1, 3, LENGTH(@var1)-3), 2, 10) AS UNSIGNED);", true, "LOAD DATA LOCAL INFILE '/tmp/t.csv' INTO TABLE `t` FIELDS TERMINATED BY 'ab' ENCLOSED BY 'b' ESCAPED BY ''"},
+		{"load data local infile '/tmp/t.csv' into table t fields terminated by 'ab' escaped by '' enclosed by 'b' SET b = CAST(CONV(MID(@var1, 3, LENGTH(@var1)-3), 2, 10) AS UNSIGNED)", true, "LOAD DATA LOCAL INFILE '/tmp/t.csv' INTO TABLE `t` FIELDS TERMINATED BY 'ab' ENCLOSED BY 'b' ESCAPED BY '' SET `b`=CAST(CONV(MID(@`var1`, 3, LENGTH(@`var1`)-3), 2, 10) AS UNSIGNED)"},
+
+		{"LOAD DATA INFILE 'file.txt' INTO TABLE t1 (column1, @dummy, column2, @dummy, column3)", true, "LOAD DATA INFILE 'file.txt' INTO TABLE `t1` (`column1`,@`dummy`,`column2`,@`dummy`,`column3`)"},
+		{"LOAD DATA INFILE 'file.txt' INTO TABLE t1 (column1, @var1) SET column2 = @var1/100", true, "LOAD DATA INFILE 'file.txt' INTO TABLE `t1` (`column1`,@`var1`) SET `column2`=@`var1`/100"},
```
Comment (kennytm) [inline parser_test.go]: Please add a test case involving multiple SET clauses, e.g.

```sql
LOAD DATA INFILE 'file.txt' INTO TABLE t 
SET a = DEFAULT, b = DEFAULT, c = CURRENT_TIMESTAMP;
```

Comment (spongedu) [reply parser_test.go]: OK, I'll add related case.

**Comment by spongedu** [comment]
@kennytm @tiancaiamao  PTAL :P

**Pull request merged**